package/src/components/ChannelList/hooks/usePaginatedChannels.ts

```diff
@@ -47,19 +47,21 @@ export const usePaginatedChannels = <
   setForceUpdate,
   sort = {},
 }: Parameters<StreamChatGenerics>) => {
-  const { client } = useChatContext<StreamChatGenerics>();
   const [channels, setChannels] = useState<Channel<StreamChatGenerics>[] | null>(null);
+  const [error, setError] = useState<Error | undefined>(undefined);
   const [staticChannelsActive, setStaticChannelsActive] = useState<boolean>(false);
+  const [activeQueryType, setActiveQueryType] = useState<QueryType | null>(null);
+  const [hasNextPage, setHasNextPage] = useState<boolean>(false);
```

Comment:
also here, initally hasnextPage is true right as before? we dont know if its false.. s

Reply:
It makes more sense to be false initially. If there's more data it will be true

Reply:
Usually these kind of things are set to true initially.. also explained by vishal in the call

Reply:
But why can't it be false, this will never be true for the first fetch and can be true only if there are more data to load, right? Just asking to check if I am wrong here.

Eg: See `loadingExtraData` here - https://alexb72.medium.com/how-to-add-pagination-to-a-react-native-flatlist-ce2425d02f1a

Reply:
Aren't we receiving that info from our BE after the first fetch? I thought we had a "page_count" or such?

Reply:
This is calculated using `channelQueryResponse.length >= newOptions.limit` everytime and also initially, so even if it's false, if there's more data it will be true. I don't understand what is wrong in this, correct me if I am wrong @vanGalilea

Reply:
Technically it makes more sense to keep it false so that no logic will load more data initially no matter there's any data or not, if there's more data it will be true and the query will be called automatically

Reply:
if its false initially.. why does it query? I thought we query only if hasNextPage is true

Reply:
If you go through the code. `queryType` is `reload` for the first time, which means it should call the first 30 sets of data, then if there's more data or equal data we set the `hasNextPage` to true and next call is done to get more data. Initially if it's `false` it will be true after the first call if there's more data. What's the point in keeping it `true` since the beginning?

Reply:
Check the `useEffect` for !enableOfflineSupport case and you will get what is happening.

Reply:
@santhoshvai we set the `hasNextPage` only if after the first `queryChannels` call  is made and when there is an extra page/data (we do it by calculating `channelQueryResponse.length >= newOptions.limit`).

Wit this PR we call `queryChannels` only when needed (hasNextPage is true) and use of resources has been limited